File: backend/src/main/java/com/gxs/bank/dto/request/LoanApplyRequest.java
package com.gxs.bank.dto.request;

import jakarta.validation.constraints.*;
import lombok.*;
import java.math.BigDecimal;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class LoanApplyRequest {

    @NotBlank(message = "Loan type is required (INSTALMENT or BALANCE_TRANSFER)")
    private String loanType;

    @NotNull(message = "Amount is required")
    @DecimalMin(value = "1000", message = "Minimum loan amount is S$1,000")
    private BigDecimal amount;

    @NotNull(message = "Tenure is required")
    @Min(value = 3, message = "Minimum tenure is 3 months")
    @Max(value = 360, message = "Maximum tenure is 360 months")
    private Integer tenureMonths;

    private String loanName; // e.g. "AIYAH", "IKEA"
}

File: backend-python/com/gxs/bank/dto/request/LoanApplyRequest.py
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, Field


class LoanApplyRequest(BaseModel):
    loanType: str = Field(..., min_length=1)
    amount: Decimal = Field(..., ge=Decimal("1000"))
    tenureMonths: int = Field(..., ge=3, le=360)
    loanName: Optional[str] = None
